projects/optic/src/__tests__/integration/spec-push.test.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  test,
+  expect,
+  describe,
+  jest,
+  beforeEach,
+  afterEach,
+} from '@jest/globals';
+import {
+  runOptic,
+  setupWorkspace,
+  normalizeWorkspace,
+  setupTestServer,
+  run,
+} from './integration';
+jest.setTimeout(30000);
+
+setupTestServer(({ url, method }) => {
+  return JSON.stringify({});
+});
+
+function sanitizeOutput(out: string) {
+  return out.replace(/git:[a-zA-Z0-9]{40}/g, 'git:COMMIT-HASH');
+}
+
+describe('optic spec push', () => {
+  let oldEnv: any;
+  beforeEach(() => {
+    oldEnv = { ...process.env };
+    process.env.LOG_LEVEL = 'info';
+  });
+
+  afterEach(() => {
+    process.env = { ...oldEnv };
+  });
+
+  test('can push a spec to a repo', async () => {
+    const workspace = await setupWorkspace('spec-push/simple', {
+      repo: true,
+      commit: true,
+    });
+    process.env.OPTIC_TOKEN = '123';
+    const { combined, code } = await runOptic(
+      workspace,
+      'spec push ./spec.yml --tag env:production,the-favorite-api'
+    );
+    expect(code).toBe(0);
+    expect(
+      normalizeWorkspace(workspace, sanitizeOutput(combined))
+    ).toMatchSnapshot();
+  });
+
+  test('requires at least one tag to push', async () => {
```

Comment:
Should we relax this for general-purpose-ness of the command? I could see wanting to push a spec up and tag it later or even some situation where I'd want to push a spec from one place, push from another place, and then do a run against it.

In some ways I feel like this command is plumbing for building up all sorts of custom workflows, so I think we should keep it as flexible as possible.

Reply:
I don't have strong opinions here, I just wasnt sure how we were intending to display this and if specs would just get lost if we didn't have a tag - updating and relaxing the requirement